src/tree_cmd.cpp

```diff
@@ -398,8 +399,15 @@ CommandCost CmdPlantTree(DoCommandFlag flags, TileIndex tile, TileIndex start_ti
 	Company *c = (_game_mode != GM_EDITOR) ? Company::GetIfValid(_current_company) : nullptr;
 	int limit = (c == nullptr ? INT32_MAX : GB(c->tree_limit, 16, 16));
 
-	TileArea ta(tile, start_tile);
-	for (TileIndex current_tile : ta) {
+	std::unique_ptr<TileIterator> iter;
```

Comment:
Seeing this (seemingly) copied bit of code made me wonder whether it's better to not copy the construction of the iterator, but change it into a function that gets reused. Also has the bonus of removing a few delete statements by move all those cases to smart pointers:
https://github.com/OpenTTD/OpenTTD/compare/master...rubidium42:OpenTTD:tileiterator_create 
Not sure whether to make a separate PR for this, or whether it's easier if it's in this branch. Though now I'm seeing more places where TileIterators could use smart pointers.

Reply:
#10345 has been merged, so you should use `std::unique_ptr<TileIterator> iter = TileIterator::Create(tile, start_tile, diagonal);` now instead of these six lines.

Reply:
Thanks for that. 🙂 Updated.